lib/coverband/collectors/coverage.rb

```diff
@@ -31,7 +31,7 @@ def reset_instance
       end
 
       def report_coverage(force_report = false)
-        return Background.start if @background_reporting_enabled
+        return Background.start if @background_reporting_enabled && !force_report
```

Comment:
@danmayer definitely needed. I don't think I tested the actual report generation enough and was more concerned with how and when the background thread was running. Sorry about that. Should I backport your fix for a 3.02 release?

Reply:
Should we consider kicking off Background.start from the middleware instead of here? Then we don't have to worry about the force_report flag.

Reply:
yeah that makes sense, perhaps back to the idea of 2 distinct middlewares... one just to start background and the other for middleware reporting?

I am wondering if we could have middleware the removes itself after it kicks off background start so we don't have it adding any overhead after that starts.

Reply:
I think we still need force report to bypast the reporting frequency...

I guess I wouldn't worry about a 3.0.2 release since we didn't have this enabled by default or fullly documented I don't think it is impacting anymore if we get a bug report we can point them towards this. We might get 4.0.0 done before it is an issue.

Reply:
Ahhh.. totally didn't think about reporting frequency. Removing all of these configuration options will make the logic much easier to reason about. Does it make sense to target removing reporting_frequency altogether considering we are now reporting from the background thread?

Agreed on not releasing a 3.0.2. Let's push for 4.0.0.

Reply:
ok take a look at latest commits @kbaum I refectored this a bit and moved background start to the middleware